app/backend/lib/gis-upload.ts

```diff
@@ -0,0 +1,106 @@
+import { Router } from 'express';
+import formidable from 'formidable';
+import Ajv from 'ajv';
+import fs from 'fs'; 
+import RateLimit from 'express-rate-limit';
+import schema from './gis-schema.json'; 
+import { performQuery } from './graphql';
+import getAuthRole from '../../utils/getAuthRole';
+
+const limiter = RateLimit({
+  windowMs: 1*60*1000, // 1 minute
+  max: 5
+});
+
+const saveGisDataMutation = `
+  mutation gisUploadMutation($input: JSON) {
+    createGisData(input: {gisData: {jsonData:$input}}) {
+      clientMutationId
+    }
+  }
+`;
+
+const gisUpload = Router();
+const ajv = new Ajv({ allErrors: true, removeAdditional: true });
+
+const formatAjv = (data, errors) => {
+  const reply = [];
+  errors.forEach((e) => {
+    const parts = e.instancePath.split('/');
+    if (parts.length > 2) {
+      const item = {
+        line: parseInt(parts[1], 10) + 1,
+        ccbc_number: data[parts[1]].ccbc_number,
+        message: `${parts[2]} ${e.message}`,
```

Comment:
Instead of `parts[1]` and `parts[2]` can we make a constant? Like `parts[ERROR_LOCATION_KEY]` instead of 1 (I'm assuming here)